core/domain/state_domain.py

```diff
@@ -2052,6 +2333,15 @@ def validate(self) -> None:
                 'Expected needs_update to be a bool, received %s' %
                 self.needs_update)
 
+        # Validate translations.
+        if isinstance(self.translation, list):
+            for html in self.translation:
+                validate_rte_tags(html)
+                validate_tabs_and_collapsible_rte_tags(html)
```

Comment:
Is it possible that translations can be unicode strings and not necessarily HTML strings? In that case, do we still have to validate tags?

Reply:
I'm not completely sure about this particular part. Also, I'm curious on how we can detect if the incoming data is unicode or HTML. And I don't think that we can validate unicode strings

Reply:
That is not true, we store different formats of translations (see "DATA_FORMAT_TO_TRANSLATABLE_OBJ_TYPE" and related constants in this class.)

The function in the previous comment points to a method that runs a conversion function on all HTML content in WrittenTranslations -- it does not imply that all translations stored are of HTML type.

Reply:
Thank you so much for the info, I get it now. Thanks. I have updated the code